### PR TITLE
[subset] Use vector instead of map for glyph bounds

### PR DESCRIPTION
**Pull request by behdad**
Is faster.

Part of https://github.com/harfbuzz/harfbuzz/issues/4231

### CODE REVIEW

File: src/hb-ot-hmtx-table.hh [hunk 1]
```diff
@@ -125,9 +126,10 @@ struct hmtxvmtx
         int lsb = _.second.second;
         max_adv = hb_max (max_adv, adv);
 
-        if (bounds_map->has (gid))
+        if (bounds_vec[gid])
         {
-          unsigned bound_width = bounds_map->get (gid);
+	  empty = false;
```
Comment (qxliu76) [inline src/hb-ot-hmtx-table.hh:131]: oh there's a composite glyph which contains only one child glyph that is EMPTY, so bounds_vec value for parent glyph is 0 and min_lsb/min_rsb/max_extent are thus not updated accordingly.

Comment (qxliu76) [reply src/hb-ot-hmtx-table.hh:131]: maybe changing the vector to store another flag indicating whether this glyph is empty or not?

Comment (qxliu76) [reply src/hb-ot-hmtx-table.hh:131]: I tried to initialize the bounds_vec value to be 0xFFFFFFFF which seems working:

```
for (unsigned i = 0; i < _num_output_glyphs; i++)
    bounds_width_vec[i] = 0xFFFFFFFF;

```

not sure if that is efficient.

Comment (behdad) [reply src/hb-ot-hmtx-table.hh:131]: Can you please push a commit to this branch for that?

Comment (qxliu76) [reply src/hb-ot-hmtx-table.hh:131]: I got this error when trying to push to upstream: 
remote: Permission to harfbuzz/harfbuzz.git denied to qxliu76.
fatal: unable to access 'https://github.com/harfbuzz/harfbuzz.git/': The requested URL returned error: 403

I pushed to googlefonts instead: https://github.com/googlefonts/harfbuzz/tree/fix-vectors

**Pull request merged**